br/pkg/task/restore.go

```diff
@@ -820,6 +823,19 @@ func runRestore(c context.Context, g glue.Glue, cmdName string, cfg *RestoreConf
 		return errors.Annotate(berrors.ErrRestoreInvalidBackup, "contain tables but no databases")
 	}
 
+	if cfg.CheckRequirements {
+		err := utils.WithRetry(ctx, func() error {
+			err = checkDiskSpace(ctx, mgr, files, tables)
+			if err != nil {
+				return errors.Trace(err)
+			}
+			return nil
+		}, utils.NewPDReqBackoffer())
```

Comment:
Not all errors can be retry. I think we'd better retry when we are really requesting PD or just remove the retry as we are at the very beginning and can easily restart the whole procedure.

Note: `NewPDReqBackoffer()` will retry almost every error because it retry when `status.Code(err)` is `codes.Unknown`, which is the default return value when `err` isn't a gRPC status error. 
https://github.com/pingcap/tidb/blob/4272571aacbf3891fcbaf8167e4e1bf99eed682a/br/pkg/utils/backoff.go#L256-L259